app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -46,10 +46,17 @@ def self.raw_connect(hostname, port, token, alerts = false)
     end
 
     def connect(options = {})
-      @client ||= self.class.raw_connect(hostname,
-                                         port,
-                                         authentication_token('default'),
-                                         options[:alerts])
+      @clients ||= {}
+      @clients[options[:alerts] ? :alerts : :metrics] ||= self.class.raw_connect(
```

Comment:
Honestly I'm not a fan of creating variables that are not used.
(not a performance thing, only readability)

If you feel current code is not readable I suggest:
```
@clients ||= {}
@clients[
  options[:alerts] ? :alerts : :metrics
] ||= self.class.raw_connect(
  hostname,
  port,
  authentication_token('default'),
  options[:alerts]
)
```

What do you think?